ui/src/app/views/workspace/component.ts

```diff
@@ -32,7 +35,22 @@ export class WorkspaceComponent implements OnInit {
     this.cohortsService.getCohortsInWorkspace('123', '123').subscribe(
           cohortsReceived => {
             this.cohortList = cohortsReceived;
+            this.CohortEditService.list().then(
+              cohorts => {
+                for (const coho of cohorts) {
+                  this.cohortList.push(coho);
+                }
+              }
+            );
           });
     this.currentUrl = this.router.url;
   }
+
+  addCohort(): void {
+    this.router.navigate([this.currentUrl + '/cohort/' + this.CohortEditService.add()]);
```

Comment:
I believe that is correct, but I'm not sure. If the name is planned to be immutable, I think that makes sense, but if the name is mutable and the ID isn't, I don't see a reason the ID should be based on the name, in which case I don't see a reason to wait to allocate an ID.